src/lib/sqle/dictionary/dictionary.service.ts

```diff
@@ -0,0 +1,829 @@
+  system_id?: string;
+  database_id?: string;
+  table_id?: string;
+  id?: string;
+  name?: string;
+}
+
+/**
+ * FunctionType
+ * A - Aggregate
+ * B - Aggregate and statistical
+ * C - Contract function
+ * D - External stored procedure
+ * F - Scalar
+ * H - User defined method
+ * I - Internal type method
+ * L - Table operator
+ * R - Table Function
+ * S - Statistical
+ */
+
+/**
+ * TableKind for dbc.tablesvx;
+ * A - Aggregate function
+ * B - Combined aggregate and ordered analytical function
+ * C - Table operator parser contract function
+ * D - JAR
+ * E - External Stored Procedure
+ * F - Standard Function
+ * G - Trigger
+ * H - Instance or contructor method
+ * I - Join Index
+ * J - Journal
+ * K - Foreign Server Object
+ * L - User defined table operator
+ * M - Macro
+ * N - Hash index
+ * O - Table with no primary index and no partitioning
+ * P - Stored Procedure
+ * Q - Queue table
+ * R - Table function
+ * S - Ordered Analytical function
+ * T - table with a primary index or primary AMP index, partitioning or both. Or a partioned table with NoPI;
+ * U - User defined type
+ * V - View
+ * X - Authorization
+ * Y - GLOP set
+ * Z - UIF - User Installed File
+ * 1 - A DATASET schema object created by CREATE SCHEMA
+ * 2 - Functon alias object
+ */
+
+/**
+ * dbc.externalSPs
+ * dbc.sessionInfovx
+ */
+
+@Injectable()
+export class VantageDictionaryService {
+  constructor(private _queryService: VantageQueryService) {}
+
+  async getFrequencyAnalysis(
+    connection: ISQLEConnection,
+    database: string,
+    table: string,
+    column: string,
+  ): Promise<any> {
+    const requests: any[] = [];
+    const session: any = await this._queryService.createSession(connection).toPromise();
+    const payload: IQueryPayload = {
+      query: `
+        call demouser.td_analyze('frequency', 'database=${database};tablename=${table};columns=${column};');
+      `,
+      session: session.sessionId,
+      rowLimit: 10000,
+    };
+    requests.push(this._queryService.querySystem(connection, payload));
+
+    const resultSets: IQueryResultSet[] = [];
+    try {
+      for (const request of requests) {
+        resultSets.push(await request.toPromise());
+      }
+    } finally {
+      await this._queryService.deleteSession(connection, session.sessionId).toPromise();
+    }
+    const rows: any[] = resultSets[0].results[0].data.map((row: any) => {
+      return {
+        value: row.xval,
+        percent: row.xpct,
+      };
+    });
+    return {
+      database,
+      table,
+      column,
+      frequency: rows,
+    };
+  }
+
+  async getValuesAnalysis(connection: ISQLEConnection, database: string, table: string, column: string): Promise<any> {
+    const requests: any[] = [];
+    const session: any = await this._queryService.createSession(connection).toPromise();
+    const payload: IQueryPayload = {
+      query: `
+        call demouser.td_analyze('values', 'database=${database};tablename=${table};columns=${column};');
```

Comment:
Ditto here. Not sure what should be in it's place, though. Nickname?

Reply:
Remove also